.github/workflows/test-generated-project.yml

```
@@ -0,0 +1,56 @@
+name: Test Generated Project
+
+on:
+  pull_request:
+    types: [opened, synchronize]
+
+env:
+  TERRAFORM_VERSION: "1.5.5"
+  TFSEC_VERSION: "v1.28.1"
+
+jobs:
+  test:
+    name: Run Tests Generated Project
+    strategy:
+      matrix:
+        node-version: [16.x, 18.x]
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout the repository
+        uses: actions/checkout@v4
+
+      - name: Setup Node
+        uses: actions/setup-node@v4
+        with:
+          node-version: ${{ matrix.node-version }}
+          cache: 'npm'
+
+      - name: Cache Node npm
+        uses: actions/cache@v3
+        with:
+          path: node_modules
+          key: ${{ runner.os }}-infrastructure-templates-node-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-infrastructure-templates
+
+      - name: Install modules
+        run: npm ci
```

```suggestion
        id: cache-nodemodules
        uses: actions/cache@v3
        with:
          path: node_modules
          key: ${{ runner.os }}-infrastructure-templates-node-${{ hashFiles('**/package-lock.json') }}
          restore-keys: |
            ${{ runner.os }}-infrastructure-templates

      - name: Install modules
        if: steps.cache-nodemodules.outputs.cache-hit != 'true'
        run: npm ci
```

That's optional, but if we give the step an id, we can refer to its result and skip the `npm ci` command when cache was found.
- PROS: faster tests run when cache is recovered
- CONS: test might fail when `package.lock.json` is not updated (e.g. a dev changed a version number in `package.json` but did not run `npm i` before committing)

I personally recommend it as that `cons` is a good way to enforce updated lock files 😇

---

Nice one, it was duplicated from the current `test.yml` workflow. Updated them both in [96f982d](https://github.com/nimblehq/infrastructure-templates/pull/276/commits/96f982d273f4cf364e0db16f2e8db51181a2d997)